src/test/java/nextstep/courses/domain/EnrollmentConditionTest.java

```diff
@@ -0,0 +1,58 @@
+package nextstep.courses.domain;
+
+import nextstep.courses.domain.session.*;
+import nextstep.courses.exception.ExceedSessionCapacityException;
+import nextstep.courses.exception.MismatchSessionFeeException;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static nextstep.courses.domain.fixture.SessionFixture.payment;
+import static nextstep.courses.domain.fixture.SessionFixture.session;
+import static nextstep.courses.exception.SessionExceptionMessage.CAPACITY_EXCEED;
+import static nextstep.courses.exception.SessionExceptionMessage.PAYMENT_MISMATCH;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.assertj.core.api.Assertions.assertThatNoException;
+
+public class EnrollmentConditionTest {
+
+    @Test
+    @DisplayName("[성공] 강의 수강에 제약이 없다.")
+    void 강의_무조건() {
+        EnrollmentCondition condition = new NoneCondition();
+        assertThatNoException()
+                .isThrownBy(() -> condition.isSatisfied(session()));
+    }
+
+    @Test
+    @DisplayName("[성공] 강의 최대 수강 인원을 초과할 수 없다.")
+    void 강의_수강인원() {
+        EnrollmentCondition condition = new CapacityCondition();
+        assertThatNoException()
+                .isThrownBy(() -> condition.isSatisfied(session(new SessionCapacity(10))));
+    }
+
+    @Test
+    @DisplayName("[실패] 최대 수강인원을 초과하면 ExceedSessionCapacityException 예외가 발생한다.")
+    void 강의_수강인원_초과() {
+        EnrollmentCondition condition = new CapacityCondition();
+        assertThatExceptionOfType(ExceedSessionCapacityException.class)
+                .isThrownBy(() -> condition.isSatisfied(session(new SessionCapacity(10, 10))))
+                .withMessageContaining(CAPACITY_EXCEED.getMessage());
+    }
+
+    @Test
+    @DisplayName("[성공] 수강생이 결제한 금액과 수강료가 일치할 때 수강 신청이 가능하다.")
+    void 강의_수강료() {
+        assertThatNoException()
+                .isThrownBy(() -> (new FeeCondition(payment())).isSatisfied(session()));
```

Comment:
예외를 검증할 땐 테스트 대상 메서드만 람다식 내부에 포함하면 어떨까요?
```suggestion
        final Payment payment = payment();
        final EnrollmentCondition condition = new FeeCondition(payment);
        final Session session = session();
        assertThatNoException()
                .isThrownBy(() -> condition.isSatisfied(session));
```

Reply:
참조하는 인스턴스가 많아지니 가이드 주신 대로 분리하는게 가독성이 더 좋네요 🙊
위에 피드백 주신 `SessionEnrollment` 분리하며 테스트 코드 전체적으로 수정하면서 반영했습니다!

의견 상세하게 공유 주셔서 감사해요 !!